src/Util/PaginationHelper.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Util;
+
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * @author Jonathan Doelfs <jd@sodatech.com>
+ */
+class PaginationHelper
+{
+    private $requestStack;
+    private $resourceMetadataFactory;
+    private $resourceMetadata;
+
+    private $enabled = true;
+    private $clientEnabled = false;
+    private $clientItemsPerPage = false;
+    private $itemsPerPage = 30;
+    private $maximumItemsPerPage = null;
+
+    private $parameterNamePage = 'page';
+    private $parameterNameEnabled = 'pagination';
+    private $parameterNameItemsPerPage = 'itemsPerPage';
+
+    private $resourceClass;
+    private $operationName;
+
+    public function __construct(RequestStack $requestStack, ResourceMetadataFactoryInterface $resourceMetadataFactory, string $resourceClass, string $operationName = null, bool $enabled = true, bool $clientEnabled = false, bool $clientItemsPerPage = false, int $itemsPerPage = 30, string $parameterNamePage = 'page', string $parameterNameEnabled = 'pagination', string $parameterNameItemsPerPage = 'itemsPerPage', int $maximumItemsPerPage = null)
```

Comment:
May I suggest an alternative approach?

* Remove the factory
* Register `PaginationHelper` (maybe can we find a more explicit name) as a service
* Pass `enabled`, `clientEnabled` and so on has parameters of methods (instead of the constructor) to make this service stateless

WDYT?